### PR TITLE
Purchasing contracts

### PR DESCRIPTION
**Pull request by cpacia**
PR adds ability to purchase contracts. Also refactors a lot of code to support running the server on testnet and running as a unix daemon.

### CODE REVIEW

File: api/wsclient.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def onMessage(self, payload, isBinary):
 
 if __name__ == '__main__':
 
-    factory = WebSocketClientFactory("ws://localhost:18466")
+    factory = WebSocketClientFactory("ws://bitcoinauthenticator.org:18466")
```
Comment (hoffmabc) [inline api/wsclient.py]: I'm assuming for testing purposes?

File: constants.py [hunk 2]
```diff
@@ -12,3 +12,5 @@
     os.makedirs(DATA_FOLDER + "store/listings/in progress/")
     os.makedirs(DATA_FOLDER + "store/listings/trade receipts/")
     os.makedirs(DATA_FOLDER + "store/media/")
+    os.makedirs(DATA_FOLDER + "purchases/in progress/")
```
Comment (hoffmabc) [inline constants.py]: Probably no spaces in folder names is safer especially for non-Windows machines.

File: db/datastore.py [hunk 3]
```diff
@@ -28,6 +28,12 @@ def create_database(filepath=None):
     cursor.execute('''CREATE TABLE moderators(guid BLOB UNIQUE, signedPubkey BLOB, encryptionKey BLOB,
 encryptionSignature BLOB, bitcoinKey BLOB, bitcoinSignature BLOB, handle TEXT)''')
     cursor.execute('''CREATE INDEX idx2 ON moderators(guid);''')
+    cursor.execute('''CREATE TABLE purchases(id BLOB UNIQUE, title TEXT, timestamp INTEGER, btc FLOAT,
```
Comment (hoffmabc) [inline db/datastore.py]: Why is the id field a BLOB? Seems overzealous doesn't it?

File: requirements.txt [hunk 4]
```diff
@@ -9,3 +9,4 @@ gnupg==2.0.2
 pynacl==0.3.0
 txrestapi==0.2
 autobahn==0.10.5-2
+python-obelisk
```
Comment (hoffmabc) [inline requirements.txt]: Which version?

Comment (cpacia) [inline requirements.txt]: 0.1.3 but I don't think that is the one on pip

File: api/wsclient.py (hunk 1 above)
Comment (cpacia) [inline api/wsclient.py]: yeah that file can be deleted after the client is functional

File: db/datastore.py (hunk 3 above)
Comment (cpacia) [inline db/datastore.py]: It's the contract ID, which is supposed to be raw bytes. I'll check.

**Pull request merged**